spec/_attachments/ic.did

```diff
@@ -121,6 +123,13 @@ service ic : {
     settings : canister_settings;
     sender_canister_version : opt nat64;
   }) -> ();
+  upload_chunk : (record {
+    canister_id : principal;
+    chunk : blob;
+  }) -> (chunk_hash);
+  stored_chunks: (record {canister_id : canister_id}) -> (vec chunk_hash);
+  delete_chunks: (record {canister_id : canister_id; hash_list : vec chunk_hash}) -> ();
+  clear_chunk_store: (record {canister_id: canister_id}) -> ();
```

Comment:
Why have this as an additional method if you can achieve the same via `delete_chunks`?

Reply:
clear_chunk_store is the poor man's garbage collection and this is when the storage used is reduced to 0. 
to avoid moving stuff around, delete_chunks doesn't really shrink the space occupied in storage (it only marks space as available), so canisters need to pay for the largest amount of storage used since the last clear_chunk_store. 
One could argue that the same can be achieved by deleting all chunks in memory (and we could detect that) but it it's probably easier to explain the difference in charging if there are also two different methods

Reply:
> to avoid moving stuff around, delete_chunks doesn't really shrink the space occupied in storage (it only marks space as available), so canisters need to pay for the largest amount of storage used since the last clear_chunk_store.

I find that very confusing given that you can reactivate the deleted chunk using its hash.

Reply:
If we would continue charging for the storage anyway, then why would one delete them in the first place?

Reply:
> I find that very confusing given that you can reactivate the deleted chunk using its hash.

What do you mean by "reactivate"? Once the chunk is deleted, you can no longer retrieve its contents at all. To do so you'd need to re-upload the same chunk again.

> If we would continue charging for the storage anyway, then why would one delete them in the first place?

This would be useful for cases where multiple Wasms are being stored and you want to replace one without deleting them all. E.g. if we want to keep the last three versions of our canister then we could delete the oldest one before inserting a new one and the memory usage would stay roughly constant.

> but I think "mark_chunk_as_unused" is more descriptive of the actual behavior.

I'm not so sure about that - the chunk **is** being deleted in the sense that no one can access it anymore. It's just that the underlying space isn't being freed. This is the same behavior you'd see when a canister "deletes" data from a structure in stable memory. However I do agree that we should emphasize that the only way stop paying for storage is via the `clear_chunk_store` method. Maybe that one should have a name which makes in clearer that it's returning the space to the subnet?

Reply:
If we want to keep the currently intended behavior, then I'd point out that `delete_chunks` does not reduce the canister's memory usage while `clear_chunk_store`. It'd be easier to specify this precisely once this other [PR](https://github.com/dfinity/interface-spec/pull/221) is merged in and the canister's memory usage is introduced into the specification.